lib/puppet/type/mount.rb

```diff
@@ -293,5 +297,18 @@ def value(name)
       dependencies[0..-2]
     end
 
+    # Autorequire the mount point's file resource
+    autorequire(:file) { Pathname.new(self[:name]) }
+
+    # Autobefore the mount point's child file paths
+    autobefore(:file) do
+      dependencies = []
+      file_resources = catalog.resources.collect { |resource| resource.type == :file }
```

Comment:
Doh, sorry about that! `collect` and `map` are actually the same method: https://github.com/ruby/ruby/blob/f98c9a0aa63c5db16005709eff807dcad525bf57/array.c#L6098-L6101, and `select` is the different one.